### PR TITLE
Support for autofill for city when entering zipcode

### PR DESCRIPTION
**Pull request by teodormoq**

### CODE REVIEW

File: packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js [hunk 1]
```diff
@@ -99,6 +104,30 @@ export class MyAccountAddressForm extends FieldForm {
...
+        const { countryId } = this.state;
+
+        try {
+            fetch(`http://api.zippopotam.us/${countryId}/${value}`)
+                .then(
+                /** @namespace Sofacompany/Route/Checkout/Container/fetch/then */
+                    (response) => response.json()
+                )
+                .then(
+                    /** @namespace Sofacompany/Route/Checkout/Container/fetch/then/then */
```
Comment (yeegor) [inline packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js:118]: Invalid namespaces here

File: packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js [hunk 2]
```diff
@@ -99,6 +104,30 @@ export class MyAccountAddressForm extends FieldForm {
...
+                    (data) => {
+                        if (data && Object.entries(data).length > 0) {
+                            this.setState({
+                                city: data.places[0].['place name']
+                            });
+                        }
+                    }
```
Comment (yeegor) [inline packages/scandipwa/src/component/MyAccountAddressForm/MyAccountAddressForm.component.js:125]: No data fetches are allowed in the components. If you wish to perform a data fetch on React level - you do that in the container.
But, consider moving this data retrieval into some util function and calling it from there. Who knows where we'll need this in the future :) Let's keep it as reusable as possible.

**Review by yeegor — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/util/Address/index.js [hunk 3]
```diff
@@ -87,3 +87,13 @@ export const getFormFields = (fields, addressLinesQty) => {
 
     return setAddressesInFormObject(fields, addressLinesQty);
 };
+
+export const getCityFromZipcode = (countryId, value) => {
```
Comment (yeegor) [inline packages/scandipwa/src/util/Address/index.js:91]: Missing namespace here

File: packages/scandipwa/src/util/Address/index.js [hunk 4]
```diff
@@ -87,3 +87,13 @@ export const getFormFields = (fields, addressLinesQty) => {
...
+    fetch(`http://api.zippopotam.us/${countryId}/${value}`)
+        .then(
```
Comment (yeegor) [inline packages/scandipwa/src/util/Address/index.js:93]: Missing namespace also here, and below.

**Review by yeegor — CHANGES_REQUESTED** [review]
Minor change necessary. Use eslint plugin to provide the namespace, please.

**Review by yeegor — APPROVED** [review]
Legit! Thank you for this contribution!

**Comment by aleksandrakorolova** [comment]
Hi @teodormoq 
I tested this functional but autofill doesn't work for me. Maybe I did something wrong or something missing
I entered zip code in form and nothing happens. Can you help me?

**Comment by teodormoq** [comment]
> Hi @teodormoq
> I tested this functional but autofill doesn't work for me. Maybe I did something wrong or something missing
> I entered zip code in form and nothing happens. Can you help me?

Hi @aleksandrakorolova do you have a url i can test on?

**Comment by aleksandrakorolova** [comment]
@teodormoq https://master.d25kd40decjpue.amplifyapp.com/

**Comment by teodormoq** [comment]
> @teodormoq https://master.d25kd40decjpue.amplifyapp.com/

Hi @aleksandrakorolova 
yes i can see that now, we need to change:
http://api.zippopotam.us/${countryId}/${value}
to
https://api.zippopotam.us/${countryId}/${value}

so its with https instead of only http.
Currently i dont really have time to change it, can you get someone to fix this? :)

**Comment by aleksandrakorolova** [comment]
@teodormoq 
Ok, I will create task for this
Thank you

**Pull request merged**